src/MarketPlace.sol

```diff
@@ -120,22 +127,28 @@ contract MarketPlace {
             s
         );
 
-        {
-            // assign values for the principal tokens and adapters array
-            for (uint i = 0; i < t.length; i++) {
-                markets[u][m].tokens[i + 1] = t[i];
-                markets[u][m].adapters[i + 1] = a[i];
+        {   
+            market.adapters = new address[](a.length);
+            // Assign values for the principal tokens and adapters array
+            for (uint i = 0; i < a.length; i++) {
+                market.tokens[i + 1] = t[i];
+                market.adapters[i] = a[i]; 
+                // TODO: Get a small review here on the logic -- The idea is we input adapter[0] as an illuminate adapter, and token is already set on line 120
```

Comment:
I'm not sure if I like this... You basically have to know that the adapters array you send to the method needs to have all 9 principals, but the principal tokens array needs to skip the first one. To me this feels slightly off and it's not something that's easy to spot.
In the previous version I noticed, we actually didn't set the pt and adapter for illuminate at all. How did they get set? And we stored the result of `ICreator(creator).create` as the market's `pool`. Now we store it as the first pt but we don't store the pool?

Reply:
Honestly there were a few errors in here from the transition that sourabh had done --

What ive left it at for now leaves this sort of weirdness where the input has different lengths, so im thinking about adding a setter for the illuminate adapter so those can always be skipped.